crates/catalog/rest/src/catalog.rs

```diff
@@ -956,6 +983,68 @@ mod tests {
         );
     }
 
+    #[tokio::test]
+    async fn test_get_default_headers() {
+        let server = Server::new_async().await;
+        let mut props = HashMap::new();
+        props.insert("credential".to_string(), "client1:secret1".to_string());
+
+        let config = RestCatalogConfig::builder()
+            .uri(server.url())
+            .props(props)
+            .build();
+        let headers: HeaderMap = config.get_default_headers().unwrap();
+
+        let expected_headers = HeaderMap::from_iter([
+            (
+                header::CONTENT_TYPE,
+                HeaderValue::from_static("application/json"),
+            ),
+            (
+                HeaderName::from_static("x-client-version"),
+                HeaderValue::from_static(ICEBERG_REST_SPEC_VERSION),
+            ),
+            (
+                header::USER_AGENT,
+                HeaderValue::from_str(&format!("iceberg-rs/{}", CARGO_PKG_VERSION)).unwrap(),
+            ),
+        ]);
+        assert_eq!(headers, expected_headers);
+    }
+
+    #[tokio::test]
+    async fn test_get_default_headers_with_custom_headers() {
+        let server = Server::new_async().await;
+        let mut props = HashMap::new();
+        props.insert("credential".to_string(), "client1:secret1".to_string());
+        props.insert(
+            "header.content-type".to_string(),
+            "application/yaml".to_string(),
```

Comment:
`header.content-type` (with `header.` prefix) is a custom header. This unit test will verify the `content-type` is in `get_default_headers()`, which will be used to create rest catalog client.

Reply:
Looks like the header config will overwrite the default one. Are we going to allow that? We may not overwrite these default headers since it makes less sense. For example, we may never customize ICEBERG_REST_SPEC_VERSION, it should come from the lib itself.  In this PR,  can we focus on allowing users to add extra headers in this PR. cc @liurenjie1024

Reply:
@flyrain I had the same question, so I end up reference [how customize header is supported in iceberg-python](https://github.com/apache/iceberg-python/blob/main/pyiceberg/catalog/rest.py#L482-L491), which allows overwrite default header.

I am happy to address that. To just confirm, pros like `header.x-client-version` (conflict with default header) should be ignored instead of overwrite default header. Is that correct? If so, should we also update python client?

Reply:
We can fix it here, and file issues in python/java repo. I assume all default headers should not be overwritten. WDYT? @whynick1 @liurenjie1024 @Fokko

Reply:
Addressed with f16f1a992ccc2d32f3a55fec23d75fb50a755640. cc @flyrain

Reply:
> We can fix it here, and file issues in python/java repo

Filed issue https://github.com/apache/iceberg-python/issues/550. Feel free to assign to me. Thanks.

Reply:
We don't want to override default headers.